core/app/[locale]/(default)/(faceted)/category/[slug]/page.tsx

```diff
@@ -77,297 +49,219 @@ const createCategorySearchParamsCache = cache(async (props: Props) => {
-  );
-  const refinedFacets = refinedSearch.facets.items.filter(
-    (facet) => facet.__typename !== 'CategorySearchFilter',
-  );
-
-  const transformedFacets = await facetsTransformer({
-    refinedFacets,
-    allFacets,
-    searchParams: { ...searchParams, ...parsedSearchParams },
-  });
-
-  const filters = transformedFacets.filter((facet) => facet != null);
-  const subCategoriesFilters = await getSubCategoriesFilters(props);
-
-  return [...subCategoriesFilters, ...filters];
-}
-
-async function getSortLabel(): Promise<string> {
-  const t = await getTranslations('Faceted.SortBy');
 
-  return t('sortBy');
-}
-
-async function getSortOptions(): Promise<SortOption[]> {
-  const t = await getTranslations('Faceted.SortBy');
-
-  return [
-    { value: 'featured', label: t('featuredItems') },
-    { value: 'newest', label: t('newestItems') },
-    { value: 'best_selling', label: t('bestSellingItems') },
-    { value: 'a_to_z', label: t('aToZ') },
-    { value: 'z_to_a', label: t('zToA') },
-    { value: 'best_reviewed', label: t('byReview') },
-    { value: 'lowest_price', label: t('priceAscending') },
-    { value: 'highest_price', label: t('priceDescending') },
-    { value: 'relevance', label: t('relevance') },
-  ];
-}
-
-async function getPaginationInfo(props: Props): Promise<CursorPaginationInfo> {
-  const search = await getRefinedSearch(props);
+  const categoryId = Number(slug);
 
-  return pageInfoTransformer(search.products.pageInfo);
-}
+  const { category } = await getCategoryPageData(categoryId);
 
-async function getShowCompare(props: Props) {
-  const { slug } = await props.params;
+  if (!category) {
+    return notFound();
+  }
 
-  const variables = cachedCategoryDataVariables(slug);
-  const data = await getCategoryPageData(variables);
+  const { pageTitle, metaDescription, metaKeywords } = category.seo;
 
-  return data.settings?.storefront.catalog?.productComparisonsEnabled ?? false;
+  return {
+    title: pageTitle || category.name,
+    description: metaDescription,
+    keywords: metaKeywords ? metaKeywords.split(',') : null,
+  };
 }
 
-const cachedCompareProductIds = cache(async (props: Props) => {
-  const searchParams = await props.searchParams;
+export default async function Category(props: Props) {
+  const { slug, locale } = await props.params;
 
-  const compareLoader = createCompareLoader();
+  setRequestLocale(locale);
 
-  const { compare } = compareLoader(searchParams);
+  const t = await getTranslations('Faceted');
 
-  return { entityIds: compare ? compare.map((id: string) => Number(id)) : [] };
-});
+  const categoryId = Number(slug);
 
-async function getCompareProducts(props: Props) {
-  const compareIds = await cachedCompareProductIds(props);
+  const { category, settings, categoryTree } = await getCategoryPageData(categoryId);
 
-  const products = await getCompareProductsData(compareIds);
+  if (!category) {
+    return notFound();
+  }
```

Comment:
If we are not using the customer access token for this query, a shopper might get a 404 page if they are logged in navigating to a category gated by a customer group. (looks like you removed it in `page-data.ts`)

Reply:
>would the result be different in the GraphQL response based on the customer access token?

Yeah it would. Let's say I have a Guest Shopper, a Logged in Shopper under customer group A, and a Logged in Shopper under customer group B. Guest Shoppers can see categories W and X; Customer Group A can see categories W, X, and Y; Customer Group B can see categories W, X, Y, and Z. Depending on the access each customer group has they can see different pages so really we need to precompute based on customer groups here.

Reply:
Heard.

> Depending on the access each customer group has they can see different pages so really we need to precompute based on customer groups here.

If I'm understanding what you're saying, we don't know the customer group until we send the request and its implicitly encoded in the CAT. So, the flag would probably be a "is anonymous"/"is authenticated" flag vs a "customer group" flag